pkg/engine/context/context.go

```diff
@@ -110,6 +110,9 @@ type Interface interface {
 
 	// AddJSON  merges the json map with context
 	addJSON(dataMap map[string]interface{}) error
+
+	// CreateDeepCopy generates a new context from existing one
+	CreateDeepCopy() Interface
```

Comment:
`DeepCopy` is a verb, we should just use that as the method name without the `Create` i.e. rename `CreateDeepCopy` to `DeepCopy`.

Reply:
Does this need to be exposed as a public method in the interface?